### PR TITLE
Prevent reflecting packets back on origin endpoints

### PR DESCRIPTION
**Pull request by NaterGator**
My fix for #107

### CODE REVIEW

File: src/mavlink-router/mainloop.cpp [hunk 1]
```diff
@@ -132,12 +132,12 @@ int Mainloop::write_msg(Endpoint *e, const struct buffer *buf)
...
 
 void Mainloop::route_msg(struct buffer *buf, int target_sysid, int target_compid, int sender_sysid,
-                         int sender_compid)
+                         int sender_compid, Endpoint *src)
 {
     bool unknown = true;
 
     for (Endpoint **e = g_endpoints; *e != nullptr; e++) {
-        if ((*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid)) {
+        if (*e != src && (*e)->accept_msg(target_sysid, target_compid, sender_sysid, sender_compid)) {
```
Comment (edersondisouza) [inline src/mavlink-router/mainloop.cpp]: What if we have two different 'elements' connected on the same endpoint, each with its own sysid? I've seen this while dealing with ROS stuff - can't remember exactly the scenario now.

Comment (edersondisouza) [inline src/mavlink-router/mainloop.cpp]: Anyway, please check `accept_msg`  method on endpoint. If this change is deemed necessary anyway, I'd say that it's better keep this kind of logic on that method, than starting scattering this throughout the code.

Comment (NaterGator) [inline src/mavlink-router/mainloop.cpp]: Are you saying they are transmitting and listening on the same endpoint? What type of endpoint?

Comment (edersondisouza) [inline src/mavlink-router/mainloop.cpp]: TCP/UDP. As I said, I don't remember properly, so maybe I'm mistaken. I sure remember having different compid on the same endpoint - but not so sure about sysid behaviour at that time. Sorry for not being able to be specific...

Comment (NaterGator) [inline src/mavlink-router/mainloop.cpp]: That's ok, I'll move this check to accept_msg as requested and spend some more time running on it. Maybe it makes sense to add a flag / config option to enable or disable loopback?

Comment (lucasdemarchi) [reply src/mavlink-router/mainloop.cpp]: @NaterGator any news on this?

**Comment by lucasdemarchi** [comment]
See #118

**Pull request closed without merging**